clients/js/test/depositSol.test.ts

```diff
@@ -0,0 +1,41 @@
+import { appendTransactionMessageInstruction, pipe } from '@solana/web3.js';
+import {
+  createDefaultTransaction,
+  expectCustomError,
+  signAndSendTransaction,
+} from '@tensor-foundation/test-helpers';
+import { Mode } from '@tensor-foundation/whitelist';
+import test from 'ava';
+import {
+  PoolType,
+  TENSOR_AMM_ERROR__WRONG_POOL_TYPE,
+  getDepositSolInstruction,
+} from '../src/index.js';
+import { ONE_SOL, TestAction } from './_common.js';
+import { setupLegacyTest } from './legacy/_common.js';
+
+test('deposit SOL into a NFT pool fails', async (t) => {
```

Comment:
similarly here, shouldnt we double check here that depositing SOL into a Token/Trade pool succeeds?

Reply:
We could do that explicitly, but `setupLegacyTest` deposits SOL for all the sell tests and those test both pool types.